src/workers/cdc.ts

```diff
@@ -295,6 +299,40 @@ const onFeedChange = async (
   }
 };
 
+const getNegativeAmount = (amount: number, user: User) => {
+  const positiveAmount = Math.abs(amount);
+  const difference = user.reputation - positiveAmount;
+
+  if (difference >= 0) {
+    return amount;
+  }
+
+  const result = positiveAmount - Math.abs(difference);
+
+  return result * -1;
+};
+
+const onReputationEventChange = async (
+  con: Connection,
+  logger: FastifyLoggerInstance,
+  data: ChangeMessage<ReputationEvent>,
+) => {
+  if (data.payload.op === 'c') {
+    const entity = data.payload.after;
+    const { amount } = entity;
+    if (amount > 0) {
+      await increaseReputation(con, logger, entity.grantToId, amount);
+    } else {
+      const user = await con.getRepository(User).findOne(entity.grantToId);
+      const result = getNegativeAmount(amount, user);
+      await increaseReputation(con, logger, entity.grantToId, result);
+    }
+  } else if (data.payload.op === 'd') {
+    const entity = data.payload.before;
+    await increaseReputation(con, logger, entity.grantToId, -entity.amount);
```

Comment:
I think a single function to handle the change in reputation is the right way to go. It's quite confusing to have `increaseReputation` which doesn't handle all the different edge cases. Other developers might get confused.
Also, I believe there's no need for JavaScript here. You can do everything in a single SQL query with more simple logic. You can use the math function `max` in SQL to make sure the new reputation is not below 0